.github/workflows/python-ci.yml

```diff
@@ -21,39 +21,59 @@ jobs:
     strategy:
```

Comment:
These tests need some better names so that they can be properly distinguished. You can just adapt the naming from the coverage job.